### PR TITLE
PUBSTWO-1744 (Series titles are not imported from SIPP)

### PR DESCRIPTION
**Pull request by orosz-usgs**
Sipp process now adds to db series title if not found.
  Moved logic of setting series title from SippConversionService.buildMpPublication
to SippProcess
  Added test publication series.
  Added article and usg series sip xml test data.
  Added test SippProcessIT

### CODE REVIEW

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippConversionService.java [hunk 1]
```diff
@@ -123,13 +119,10 @@ protected MpPublication buildObjectProperties(InformationProduct informationProd
...
 
 		String largerWorkTitle = informationProduct.getJournalTitle();
-		if (StringUtils.isNotBlank(largerWorkTitle)) {
-			if (PubsUtils.isPublicationTypeArticle(mpPublication.getPublicationType())
-					&& null != mpPublication.getPublicationSubtype()) {
-				mpPublication.setSeriesTitle(buildSeriesTitle(mpPublication.getPublicationSubtype(), largerWorkTitle));
-			} else {
+		if (StringUtils.isNotBlank(largerWorkTitle) &&
+			(!(PubsUtils.isPublicationTypeArticle(mpPublication.getPublicationType())
+					&& null != mpPublication.getPublicationSubtype()))) {
```
Comment (duselmann) [inline src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippConversionService.java:124]: Since they are all "and" conditions, there are unnecessary parenthesis.
Also, a null check is more important than method call check.
Suggested replacement...


> if ( StringUtils.isNotBlank(largerWorkTitle) 
>    && null != mpPublication.getPublicationSubtype() 
>    && ! PubsUtils.isPublicationTypeArticle(mpPublication.getPublicationType()) ) {

Comment (duselmann) [reply src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippConversionService.java:124]: Humm, it will not format as I expect.

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java [hunk 2]
```diff
@@ -111,9 +111,15 @@ protected InformationProduct getInformationProduct(String ipNumber) {
 				informationProduct.setPublicationType(idsPubTypeConv.getPublicationType());
 				informationProduct.setPublicationSubtype(idsPubTypeConv.getPublicationSubtype());
 
-				informationProduct.setUsgsSeriesTitle(
-						getSeriesTitle(idsPubTypeConv.getPublicationSubtype(),
+				if (StringUtils.isNotBlank(informationProduct.getJournalTitle())
+						&& PubsUtils.isPublicationTypeArticle(idsPubTypeConv.getPublicationType())
+						&& null != idsPubTypeConv.getPublicationSubtype()) {
```
Comment (duselmann) [inline src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java]: this looks similar to what I posted prior. I would move the not null check again.
If they are identical then I would make a helper function. They look identical to me.

Comment (duselmann) [reply src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java]: The results of each condition perform mostly the same code except for the inner most actions: getJournalTitle and getUsgsSeriesType.
I would recommend moving the common code out of the conditional and placing the results of the inner most action in a variable.
Turns out multiple line code suggestions are not clean or I would post what I mean.

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java [hunk 3]
```diff
@@ -131,6 +137,7 @@ protected InformationProduct getInformationProduct(String ipNumber) {
 	}
 
 	protected PublicationSeries getSeriesTitle(PublicationSubtype pubSubtype, String text) {
+		PublicationSeries ret = null;
```
Comment (duselmann) [inline src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java:140]: "ret" is not the best variable name. maybe pubSeriesTitle or something describing what is returned.

**Review by duselmann — CHANGES_REQUESTED** [review]
Looks pretty good. To me it looks like it addresses the bug appropriately and that there are tests to demonstrate the resolution. I just have some code refinement suggestions.

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippConversionService.java (hunk 1 above)
Comment (orosz-usgs) [reply src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippConversionService.java:124]: The refactored condition is not equivalent due to de Morgan’s Theorem: !(a && b) = !a || !b
A unit test caught it.  I agree that the conditional is ugly and hard to read. Left it as close to
the original to have the minimal chance of breaking. Having || in the conditional probably still leaves it hard to follow. Likely should be a helper method. Time wise, will probably leave as is.

**Review by mbucknell — APPROVED** [review]
Like the additional and new tests!
